compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -954,8 +954,9 @@ void resolve(BindingKey bindingKey) {
           /* Resolve in the parent in case there are multibinding contributions or conflicts in some
            * component between this one and the previously-resolved one. */
           parentResolver.get().resolve(bindingKey);
-          if (!new LocalDependencyChecker().dependsOnLocalBindings(bindingKey)
-              && getLocalExplicitBindings(bindingKey.key()).isEmpty()) {
+          LocalDependencyChecker localDependencyChecker = new LocalDependencyChecker();
+          if (!localDependencyChecker.dependsOnLocalBindings(bindingKey) &&
+              !localDependencyChecker.bindingDefinedLocally(bindingKey)) {
```

Comment:
I think we can do better here, will try to merge this check into `localDependencyChecker.dependsOnLocalBindings(bindingKey)`